src/Methods.js

```diff
@@ -115,14 +115,24 @@ export default function DynamicMethods({ isDarkMode }) {
   const evmWallet = userWallets.find((wallet) => wallet.chain === 'EVM');
   const btcWallet = userWallets.find((wallet) => wallet.chain === 'BTC');
 
-  const onStakeClick = async () => {
+  const onStakeClick = useCallback(async () => {
+    if (!userHasEmbeddedWallet()) {
+      console.log('creating embedded wallet')
+      try {
+        const wallet = await createEmbeddedWallet();
```

Comment:
Function `createEmbeddedWallet` suppose to return Promise<Wallet>. If that's correct we can log it out to the console but it appears that it's not happening.